### DISCUSSION TITLE
Auth via GitHub

### DISCUSSION THREAD

**Discussion by elenashcherbinina** [hexlet-volunteers/runit#377] [Q&A] [answered]
конфигирурию GitHub App и не понимаю, куда нужно добавить environment variables for the deploy: OAUTH_CLIENT_ID и тд.

**Comment by fey** [comment]
@faciledictu Можете подсказать,  может у нас в инструкциях чего-то не хватает? @elenashcherbinina Можете описать, что делаете, какие ошибки выводятся?

**Reply by elenashcherbinina** [reply]
я сделала все, как описано в read.me. Но куда вставлять вот это все, не знаю:
OAUTH_CLIENT_ID Client ID from your GitHub App.
OAUTH_CLIENT_SECRET Generate a new Client secret in the GitHub App settings and specify it here.
OAUTH_ACCESS_TOKEN_URL=https://github.com/login/oauth/access_token
OAUTH_AUTHORIZE_URL=https://github.com/login/oauth/authorize
GITHUB_USER_URL=https://api.github.com/user

В githubApp нужен был только  Authorization callback URL. Там нет поля для environment variables.

Ошибок как таковых еще нет, так как деплой на рендер тоже упал.

**Reply by fey** [reply]
На вкладке Environment добавляются переменные. Эти переменные используюся в приложении уже.
![image](https://github.com/hexlet-rus/runit/assets/697178/f9d2414a-b763-41ef-ad3e-c10af9c33cfb)
![image](https://github.com/hexlet-rus/runit/assets/697178/fc9a6245-03ea-4168-b266-31bee7b082a5)

**Comment by elenashcherbinina** [comment]
То есть вот эти переменные нужно в рендер записать? Я просто думала, что в githubApp тоже есть какие-то variables, потому что в инструкции было так написано. Сейчас в рендер у меня вот эти ключи:
![Снимок экрана 2023-09-15 в 13 51 51](https://github.com/hexlet-rus/runit/assets/108113888/004a8c33-5714-4a47-a511-913363d3493f)

Попробую еще раз зедеплоить, и добавить новые ключи. Еще попробую способ с oAuth. Отпишусь

**Reply by fey** [reply]
Да, верно, вы эти переменные берете из гитхаба и кладете в окружение приложения.

**Comment by elenashcherbinina** [comment, marked as answer]
Итак, сделала авторизацию через oAuth. Все работает: https://elenashcherbinina-runit.onrender.com/
